doc/sg/c_cpp_coding_style.md

```diff
@@ -355,6 +355,22 @@ Macros are often necessary and reasonable coding practice C (as opposed to C++)
 In contrast to the recommendation in the Google C++ style guide, exporting macros as part of the public API is allowed in C code.
 A typical use case is a header with register definitions.
 
+**Function-like macros**
+Function-like macros should be avoided whenever possible since they are error-prone. Where they are necessary, they should be hygienic. Here are some useful tips:
+- Expand the macro arguments between brackets `()` since the caller could use an expression as argument.
```

Comment:
nit: Expand the macro arguments between brackets `()` **as** the caller could use an expression as **an** argument.